### ISSUE TITLE
Fix Spring Boot1 example dependencies

### ISSUE THREAD

**Issue by uweschaefer** [factcast/factcast#373]
_From @uweschaefer on 2019-04-10 19:48:09 +0000 UTC_

**Describe the bug**
accidentally, there were dependabot PRs merged, that shifted the dependency of 
`factcast-examples/factcast-example-client-spring-boot1` to spring boot 2

_Copied from original issue: Mercateo/factcast#373_

**Comment by uweschaefer** [comment]
_From @mweirauch on 2019-04-10 20:28:45 +0000 UTC_

I did some tinkering regarding this issue today. The thing is we might have ended up in a problematic sitatution. The `AbstractMethodError` might be due to the case that latest version of `grpc-spring-boot-starter` (2.3.0.RELEASE) or actually since 2.2.0.RELEASE is building to Spring Boot 2.1 and and thus Spring Framework 5.1.

The "last" combination working is:
* `factcast:0.0.34`
* `grpc-spring-boot-start:2.1.0.RELEASE`
* `spring-boot:1.5.20.RELEASE`

In case of starting the `factcast-example-client-spring-boot-1` I get a bean creation exception for `EventListenerMethodProcessor`. This class has seen some changes between `5.0.x` and `5.1.x` so perhaps the expected class doesn't match up what parts of the `grpc-spring-boot-starter` where compiled against. (Wild guess.)

**Comment by uweschaefer** [comment]
_From @uweschaefer on 2019-04-10 20:35:35 +0000 UTC_

thx for investigating. we should fix the last known working versions and deprecate the use of SB1. anything else sounds like a rabbit hole of pointless work compared to the well documented process of switching the client's application from SB1 to SB2.

**Comment by uweschaefer** [comment]
_From @uweschaefer on 2019-04-10 20:45:09 +0000 UTC_

damn. looks like there was an incompatible change in grpc-spring-boot-start regarding GrpcChannelFactory

**Comment by uweschaefer** [comment]
_From @mweirauch on 2019-04-10 20:47:54 +0000 UTC_

Yap. Only thing I came up with is pinning the versions in the client example (not even relying on FactCasts project.version):
```
		<factcast.version>0.0.34</factcast.version>
		<spring-grpc.version>2.1.0.RELEASE</spring-grpc.version>
		<spring-boot.version>1.5.20.RELEASE</spring-boot.version>
```

**Comment by uweschaefer** [comment]
_From @uweschaefer on 2019-04-10 21:07:12 +0000 UTC_

i don't want to go back to 0.0.34 for SB1 compatibility.

i might have a very nasty hack working for bridging the interfaces...

**Comment by uweschaefer** [comment]
_From @uweschaefer on 2019-04-10 21:10:46 +0000 UTC_

something like this will do the trick:

```
   @Bean
    GrpcChannelFactory compatibleGrpcChannelFactory(
            net.devh.springboot.autoconfigure.grpc.client.GrpcChannelFactory f) {
        return new net.devh.boot.grpc.client.channelfactory.GrpcChannelFactory() {

            @Override
            public Channel createChannel(String name, List<ClientInterceptor> interceptors) {
                return f.createChannel(name, interceptors);
            }

            @Override
            public Channel createChannel(String name) {
                return f.createChannel(name);
            }
        };
    }
```

we could have a compatibility lib here for use with <spring-grpc.version>2.1.0.RELEASE</spring-grpc.version> that provides the interface and the bean. 

rather yucky, but the only thing i can come up with right now...

**Comment by uweschaefer** [comment]
_From @uweschaefer on 2019-04-10 22:25:40 +0000 UTC_

@nln42 @mweirauch  guys, thanks for your help on finding and exploring this, really very appreciated!

could you have a look at the attached PR and verify that it works for you? see example/boot1 #374

**Comment by uweschaefer** [comment]
_From @uweschaefer on 2019-04-11 13:19:02 +0000 UTC_

closed by #374
